cmd/atlasci/internal/ci/run.go

```diff
@@ -156,3 +157,43 @@ func (f *FileReport) WriteReport(r sqlcheck.Report) {
 func (w *TemplateWriter) WriteReport(r SummaryReport) error {
 	return w.T.Execute(w.W, r)
 }
+
+// MigrationAnalyzer implements sqlcheck.Analyzer. It validates the migration dir.
+type MigrationAnalyzer struct {
+	Dir migrate.Dir
+}
+
+// Analyze implements the sqlcheck.Analyzer interface.
+func (m *MigrationAnalyzer) Analyze(_ context.Context, p *sqlcheck.Pass) error {
```

Comment:
Let's return an error instead and not continue to the rest of the analyzers. WDYT?